lib/src/components/supa_native_auth.dart

```diff
@@ -0,0 +1,205 @@
+import 'dart:async';
```

Comment:
I wonder if it's better to mix these native sign in features with the `SupaSocialsAuth` widget. 

The public API could be something like:

```dart
SupaSocialsAuth(
    socialProviders: [
        OAuthProvider.apple,
        OAuthProvider.google,
    ],
    colored: true,
    nativeGoogleAuthConfig: NativeGoogleAuthConfig(
      webClientId: 'my-google-web-client-ID',
      iosClientId: 'my-ios-google-client-ID',
    ), 
    enableNativeAppleAuth: false, // boolean where users can enable/ disable Native Apple Auth. 
    // We can probably have `true` as the default value, and users can disable it if they don't want native sign in for some reason.
    onSuccess: (Session response) {
        // do something, for example: navigate('home');
    },
    onError: (error) {
        // do something, for example: navigate("wait_for_email");
    },
),
```
We can call the native sign in flow if the user has provided the appropriate credentials. What do you think?

Reply:
Might work
But I wonder if it wont be very obvious  to the user?

Reply:
I personally think it would be more intuitive if we bundle the OAuth login with native social login. As far as the users are concerned, they are the same exact thing. Also, if we don't include them in with other social logins, the UI will be limited. 

As long as we include the instruction on the docs, the users should be able to discover it. I also think the SupaSocial widget is the one that would look for if I were using this library and wanted to add native logins!

Reply:
That's ok too. 
We can bundle them together

Reply:
ok, so I'll take it in that direction then